article/app/controllers/ArticleController.scala

```diff
@@ -234,15 +250,25 @@ class ArticleController extends Controller with RendersItemResponse with Logging
 }
 
 object ParseBlockId extends RegexParsers {
-  def apply(input: String): Option[String] = {
-    def withParser: Parser[Unit] = "with:" ^^ { _ => () }
-    def block: Parser[Unit] = "block-" ^^ { _ => () }
-    def id: Parser[String] = "[a-zA-Z0-9]+".r
-    def expr: Parser[String] = withParser ~> block ~> id
+
+  def withParser: Parser[Unit] = "with:" ^^ { _ => () }
+  def block: Parser[Unit] = "block-" ^^ { _ => () }
+  def id: Parser[String] = "[a-zA-Z0-9]+".r
+  def blockId = block ~> id
+
+  def fromPageParam(input: String): Option[String] = {
+    def expr: Parser[String] = withParser ~> blockId
 
     parse(expr, input) match {
       case Success(matched, _) => Some(matched)
       case _ => None
     }
   }
+
+  def fromBlockId(input: String): Option[SinceBlockId] = {
+    parse(blockId, input) match {
+      case Success(matched, _) => Some(SinceBlockId(matched))
+      case _ => None
+    }
+  }
```

Comment:
`fromBlockId` and `fromPageParam` have similar name and input param but different output types. Can we make them have a similar and consistant signature so when using either of them you don't have to think to much or transform the output?